src/server/login.js

```diff
@@ -14,6 +14,7 @@ module.exports = function (client, server, options) {
   } = options
 
   let serverId
+  let needToVerify
```

Comment:
this shouldn't be a top level variable, instead it should be decided on a case-by-case basis (inside the onLogin function) based on the exceptions array and the result of the login function in that specific case

Reply:
The result is required in `onEncryptionKeyResponse()` and `loginClient()` as well during the login process of a single client. Calling the function three times per login is inefficient and probably also unexpected. This is only the top level inside the function that's called individually for each client, so it's still a case-by-case basis for each client.

Reply:
I agree